From 3286baf646fa7974c165efd9b63c690d08dff6b7 Mon Sep 17 00:00:00 2001
Message-Id: <3286baf646fa7974c165efd9b63c690d08dff6b7.1518102033.git.maciej.zenon.borzecki@canonical.com>
From: Maciej Borzecki <maciej.zenon.borzecki@canonical.com>
Date: Thu, 8 Feb 2018 15:57:13 +0100
Subject: [PATCH] cmd/snap-seccomp: drop link flags that will be rejected by
 go1.9.4

Due to CVE-2018-6574 Go rejects a number of previously allowed flags cgo flags.
Drop any flags passed to pkg-config. Drop static link flags for libseccomp as
those are not supported on Arch anyway.

Signed-off-by: Maciej Borzecki <maciej.zenon.borzecki@canonical.com>
---
 cmd/snap-seccomp/main.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmd/snap-seccomp/main.go b/cmd/snap-seccomp/main.go
index 935b3098b434053808c87efe9cefe682124a09a4..ffb9bb3bcc55c9c327b4f1937bd8a86b498e74ec 100644
--- a/cmd/snap-seccomp/main.go
+++ b/cmd/snap-seccomp/main.go
@@ -20,8 +20,8 @@
 package main
 
 //#cgo CFLAGS: -D_FILE_OFFSET_BITS=64
-//#cgo pkg-config: --static --cflags libseccomp
-//#cgo LDFLAGS: -Wl,-Bstatic -lseccomp -Wl,-Bdynamic
+//#cgo pkg-config: libseccomp
+//#cgo LDFLAGS: -lseccomp
 //
 //#include <asm/ioctls.h>
 //#include <ctype.h>
-- 
2.16.1

